signature/src/image.rs

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) 2021 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::{anyhow, Result};
+use oci_distribution::Reference;
+use crate::signatures;
+
+pub const DOCKER_TRANSPORT_NAME: &str = "docker";
+#[allow(dead_code)]
+pub const DIR_TRANSPORT_NAME: &str = "dir";
+
+// Image containes informations of the image which may be used in signature verification.
+pub struct Image {
+    pub reference: Reference,
+    // digest format: "digest-algo:digest-value"
+    pub manifest_digest: String,
+
+    // Try signature extensions in registry before using the sigstore.
+    // (https://github.com/containers/image/issues/384)
+    //
+    // FIXME: now don't support, always set to false.
+    pub registry_xrss_api_support: bool,
+
+    sigstore_base_url: String,
+    signatures: Vec<Vec<u8>>,
+}
+
+impl Image {
+    pub fn default_with_reference(image_ref: Reference) -> Self {
+        Image {
+            reference: image_ref,
+            manifest_digest: String::default(),
+            registry_xrss_api_support: false,
+            sigstore_base_url: String::default(),
+            signatures: Vec::default(),
+        }
```

Comment:
This seems to be required all because oci-distribution doesn't derive `Default` on `Reference` :cry: It might be a good idea to raise a PR on that crate to allow this code to be simplified in future?